onnx/common/tensor.h

```diff
@@ -27,6 +29,115 @@ struct Tensor final {
   bool is_raw_data_;
   std::string raw_data_;
 
+  template<typename T>
+  static void add_nums(T* x, const T* y) {
+    *x += *y;
+  }
+
+  template<typename T>
+  static void sub_nums(T* x, const T* y) {
+    *x -= *y;
+  }
+
+  template<typename T>
+  static void mult_nums(T* x, const T* y) {
+    *x *= *y;
+  }
+
+  template<typename T>
+  static void divide_nums(T* x, const T* y) {
+    *x /= *y;
+  }
+
+  template<typename T>
+  static void sqrt_num(T* x) {
+    *x = (T) std::sqrt((double) *x);
+  }
+
+
+  template<typename T, int64_t block_size>
+  void bin_func(void (*f)(T*, const T*), Tensor& a, std::vector<T>& T_data_, std::vector<T>& a_T_data_) {
```

Comment:
Can we have
```
template<typename T, typename F, int64_t block_size>
void BinFunc(F&& f, Tensor* a, std::vector<T>* T_data, std::vector<T>* a_T_data)
```
as signature? First, it conforms to the Google code style better. Second, it avoids passing by function pointers. Simple functions like binary ops should really be inlined.

Reply:
This would get rid of type checks on f right? Is there a need for that?

Reply:
Compiler will do type check and throw error if substitution doesn't work. Making a function call to each of the element in a tensor is potentially costly compared to inlined.

Reply:
changed template structure to reflect above